src/json-validator.cpp

```diff
@@ -305,10 +305,10 @@ void json_validator::insert_schema(const json &input, const json_uri &id)
 	// allocate create a copy for later storage - if resolving reference works
 	std::shared_ptr<json> schema = std::make_shared<json>(input);
 
-	do {
-		// resolve all local schemas and references
-		resolver r(*schema, id);
+	// resolve all local schemas and local and external references
+	resolver r(*schema, id);
```

Comment:
Hmm, putting resolver creation outside the loop. I'm not sure whether this is a good idea concerning another round of ref-resolving when some new external refs have been updated.

Reply:
I have tested  https://github.com/pboettch/json-schema-validator/tree/fix-issue-9, unfortunately resolving of relative references is still broken.  The casue of issues are:
1. current path (path_) can be emptied, while  json_uri::update() is being called. i.e. json-uri.cpp line 103 `path_.back()` will cause a crash.
2. each time when resolver is being called, the current path of the external ref is updated, having resolver doing another round of ref-resolving, would end up constructing an invalid path for the external ref. 

Please find attached schema set, which could be easily be showing the above issues. 
[schema.zip](https://github.com/pboettch/json-schema-validator/files/2089785/schema.zip)

Test was done using json-schema-validate to load the base.json schem, and schema loader to load the other relative schema.

```
static void loader_local(const json_uri &uri, json &schema)
{
    	const std::string base("D:/DATA/tmp/schema/"); // where schema set are installed 
	std::fstream lf(base + uri.path());
	if (!lf.good())
		throw std::invalid_argument("could not open " + uri.url() + " tried with " + uri.path());
	try {
		lf >> schema;
	}
	catch (std::exception &e) {
		throw e;
	}
}
```

Reply:
Thanks for your valuable feedback, I will take a look into it.

Reply:
You're right, your version of resolving the relative path is working correctly. But there are still some issues with the ref and refRemote tests.